src/Pages/ProductListing/FiltersContainer.jsx

```diff
@@ -0,0 +1,252 @@
+import React, {useState, useEffect} from 'react';
+import useAxios from '../../Api/useAxios/useAxios';
+import { ratingsFilter } from '../../Data';
+import { getCategories } from '../../Api/actions';
+import InputField from '../../Shared/Input/InputField';
+import Input from '../../Shared/Input/Input';
+import { useProductFilters } from '../../Context/ProductContext';
+import { useParams } from 'react-router-dom';
+
+const  FiltersContainer = (props) =>{
+    
+    // let { categoryId } = useParams();
+    let { response, loading, error } = useAxios(getCategories);
+    const [data, setData] = useState([]);
+    const { state, dispatch} = useProductFilters();
+    const { sortBy , categories , price , discount , inStock, ratings} = state;
+    const [ctgrySelected, setCtgrySelected] = useState({indx:-1, ctgryId:0})
+    const [brandSearchKey , setBrandSearchKey] = useState('')
+    const [prdctTags, setPrdctTags] = useState([]);
+    const [ brands, setBrands] = useState([]);
+  
+    useEffect(()=>{
+        if(Object.keys(props.categry).length && data?.categories?.length){
+            const indx = data?.categories?.findIndex(el=> el.title === props.categry.categoryName)
+            setCtgrySelected({indx:indx, ctgryId:props.categry.categoryId})
+        }
+    }, [props])
+
+    useEffect(() => {
+        if (response !== null) {
+            setData(response);
+        }
+    }, [response]);
+
+    useEffect(() => {
+        if(data?.categories?.length && ctgrySelected.ctgryId!==0){
+            let fltrArr = data.categories.filter(el=> el.id == ctgrySelected.ctgryId)
```

Comment:
Suggestion: Can use `===` instead of `==` , it will might create problem in type comparing